File: backend/tests/DigitalFamilyCookbook.Tests.Utilities/Mocks/MockStep.cs
namespace DigitalFamilyCookbook.Tests.Utilities.Mocks;

public static class MockStep
{
    public static StepDto GenerateDto(int? sortOrder = null) => new StepDto
    {
        StepId = MockDataGenerator.RandomInteger(),
        Id = MockDataGenerator.RandomId(),
        Direction = MockDataGenerator.RandomString(200),
        SortOrder = sortOrder.HasValue ? sortOrder.Value : MockDataGenerator.RandomInteger(),
    };

    public static Step GenerateDomainModel(int? sortOrder = null) => Step.FromDto(GenerateDto(sortOrder));

    public static StepApiModel GenerateApiModel(int? sortOrder = null) => StepApiModel.FromDomainModel(GenerateDomainModel(sortOrder));

    public static List<StepDto> GenerateForRecipeDto(int stepCount = 5)
    {
        var steps = new List<StepDto>();

        for (var i = 0; i < stepCount; i += 1)
        {
            steps.Add(GenerateDto(i + 1));
        }

        return steps;
    }

    public static List<Step> GenerateForRecipe(int stepCount = 5)
    {
        var steps = new List<Step>();

        for (var i = 0; i < stepCount; i += 1)
        {
            steps.Add(GenerateDomainModel(i + 1));
        }

        return steps;
    }

    public static List<StepApiModel> GenerateForRecipeApiModel(int stepCount = 5)
    {
        var steps = new List<StepApiModel>();

        for (var i = 0; i < stepCount; i += 1)
        {
            steps.Add(GenerateApiModel(i + 1));
        }

        return steps;
    }
}

File: frontend/src/test/mocks/MockStep.ts
import DataGenerator from '@test/DataGenerator';
import Step from '@models/Step';

export const MockStep = (recipeId: number, id = 0): Step => ({
    id: DataGenerator.GenerateGuid(),
    stepId: id === 0 ? DataGenerator.GenerateInteger(1, 250) : id,
    direction: DataGenerator.GenerateString(),
    recipeId,
    sortOrder: DataGenerator.GenerateInteger(1, 20),
});

export const MockStepList = (recipeId: number, length = 10): Step[] => {
    const steps: Step[] = [];

    for (let i = 0; i < length; i += 1) {
        steps.push(MockStep(recipeId));
    }

    return steps;
};
